cdi/src/main/java/io/smallrye/concurrency/inject/SmallryeConcurrencyCdiExtension.java

```diff
@@ -251,4 +261,13 @@ private String getUniqueName(InjectionPoint ip) {
             .findFirst();
         return (optionalQulifier.isPresent()) ? optionalQulifier.get().value() : null;
     }
+
+    private boolean hasCustomQualifiers(InjectionPoint ip) {
+        // collect all qualifiers other than NamedInstance or CDI implied Default and Any
+        Set<Annotation> qualifiers = ip.getQualifiers().stream()
+                .filter(ann -> !ann.annotationType().equals(NamedInstance.class)
+                        && !ann.annotationType().equals(Default.class) && !ann.annotationType().equals(Any.class))
+                .collect(Collectors.toSet());
```

Comment:
Instead of `.filter` + `.collect` + `.isEmpty`, you can use `anyMatch` or `allMatch`.

Reply:
good point, thanks